server/rotation.go

```diff
@@ -122,13 +122,13 @@ func (k keyRotater) rotate() error {
 	}
 
 	var nextRotation time.Time
-	err = k.Storage.UpdateKeys(func(keys storage.Keys) (storage.Keys, error) {
+	err = k.Storage.UpdateKeys(func(keys storage.Keys) (*storage.Keys, error) {
 		tNow := k.now()
 
 		// if you are running multiple instances of dex, another instance
 		// could have already rotated the keys.
 		if tNow.Before(keys.NextRotation) {
-			return storage.Keys{}, nil
+			return nil, nil
```

Comment:
We should probably just define a sentinel error. E.g.

```go
errSkipKeys := errors.New("key already rotated")
```

That'll keep the UpdateKeys from updating, then we can check it later (e.g. line 168)